File: python_code/Bootstrap_new/a2_boot_figure.py
import numpy as np
import matplotlib.pyplot as plt
import os


def a2_boot_figure(GQ, X0):
    from common_codes.a7_all_mdl_prediction import a7_all_mdl_prediction
    pred1, pred2, pred3 = a7_all_mdl_prediction(X0)
    pred_all = np.column_stack([pred1, pred2, pred3])
    a = np.arange(1, pred_all.shape[0] + 1)

    base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
    save_dir = os.path.join(base_dir, 'saved_models')
    data = np.load(os.path.join(save_dir, 'Bootconf.npz'), allow_pickle=True)
    Boot_main = data['Boot_main']

    std_lin = np.std(Boot_main[0], axis=1, ddof=1)
    std_ann = np.std(Boot_main[1], axis=1, ddof=1)
    std_rf = np.std(Boot_main[2], axis=1, ddof=1)
    std_all = np.column_stack([std_lin, std_ann, std_rf])

    s = plt.figure()
    s.set_facecolor('white')
    names = ['(a)', '(b)', '(c)']

    for i in range(pred_all.shape[1]):
        ax = s.add_subplot(1, 3, i + 1)
        ax.scatter(a, pred_all[:, i], marker='o', color='b', alpha=0.7)
        ax.errorbar(a, pred_all[:, i], yerr=std_all[:, i],
                     fmt='none', capsize=3, color='r', linewidth=1)
        ax.set_title(names[i])
        ax.set_xlabel('Sample ID')
        ax.set_ylabel('Model predicted GWQI')
        ax.grid(True, alpha=0.3)

    plt.tight_layout()
    return s

File: Bootstrap_new/a2_boot_figure.m
function a2_boot_figure % Solving the system, check
close all
clearvars
clc
format short g, format compact
profile on
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
addpath('D:\Dropbox\Naresh_Sahu\Post Monsoon ANN\Codes_today\common_codes')
load b0_X_GQ.mat GQ X0
[pred1,pred2,pred3]=a7_all_mdl_prediction(X0{:,:});    % pred 1,2,3 are linear,ann and RF respectively

pred_all=[pred1,pred2,pred3];
a=1:size(pred_all,1);

load Bootconf.mat Boot_main
std_lin=std(Boot_main{1},[],2);
std_ann=std(Boot_main{2},[],2);
std_rf=std(Boot_main{3},[],2);

std_all=[std_lin std_ann std_rf];

s=figure();
set(gcf,'color','w')
name='a':'z';


for i=1:size(pred_all,2)
    subplot(1,3,i)
    h=scatter(a',pred_all(:,i),"o"); hold on
    er_bar= errorbar(a',pred_all(:,i),std_all(:,i), 'vertical', 'LineStyle', 'none','LineWidth',1);
    h.Parent.Title.String=['(' name(i) ')'];
    h.Parent.XLabel.String='Sample ID';
    h.Parent.YLabel.String='Model predicted GWQI';
    box on
end
copygraphics(s,'Resolution',600,'BackgroundColor','white','Padding','tight','PreserveAspectRatio','on')
disp('')
